client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -90,16 +93,21 @@ public void createReaderGroup(String groupName, ReaderGroupConfig config) {
         createStreamHelper(getStreamForReaderGroup(groupName), StreamConfiguration.builder()
                                                                                   .scalingPolicy(ScalingPolicy.fixed(1))
                                                                                   .build());
+        Serializer<Update<ReaderGroupState>> updateSerializer = new ReaderGroupStateUpdatesSerializer();
+        Serializer<InitialUpdate<ReaderGroupState>> initialSerializer = new ReaderGroupStateInitSerializer();
         @Cleanup
-        StateSynchronizer<ReaderGroupState> synchronizer = clientFactory.createStateSynchronizer(NameUtils.getStreamForReaderGroup(groupName),
-                                              new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
+        StateSynchronizer<ReaderGroupState> synchronizer = new ReaderGroupStateSynchronizer(groupName, initialSerializer, updateSerializer,
+                SynchronizerConfig.builder().build(), clientFactory, controller);
+        controller.createReaderGroup(groupName, config);
         Map<SegmentWithRange, Long> segments = ReaderGroupImpl.getSegmentsForStreams(controller, config);
-        synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config), 0));
     }
 
     @Override
     public void deleteReaderGroup(String groupName) {
-        getAndHandleExceptions(controller.sealStream(scope, getStreamForReaderGroup(groupName))
+        val result = getThrowingException(controller.deleteReaderGroup(groupName));
+        if (result) {
```

Comment:
This is seal and deleteStream is not needed any more.
It is done by controller.deleteReaderGroup()
Also you need to pass the ReaderGroup scope alsone with groupName in the create/delete/update ReaderGroup calls.